### PR TITLE
deposit

### PR DESCRIPTION
**Pull request by evlekht**
wip
there is a whole deposit, undeposit, system-undeposit and deposit offers impl
I'd split up PR into at least 3: deposit offers, depositTx/unlockDeposiTx, automatic depositTx from block builder
this pr in its current state is a temporal thing

### CODE REVIEW

File: genesis/genesis_kopernikus.go [hunk 1]
```diff
@@ -40,7 +40,8 @@ var (
 				SupplyCap:          1000 * units.MegaAvax,
 			},
 			CaminoConfig: config.CaminoConfig{
-				DaoProposalBondAmount: 100 * units.Avax,
+				DaoProposalBondAmount:     100 * units.Avax,
+				GradualUnlockHalfDuration: 30 * time.Second,
```
Comment (c4t-ag) [inline genesis/genesis_kopernikus.go:44]: This should go into an offer

File: vms/platformvm/blocks/builder/builder.go [hunk 2]
```diff
@@ -1,3 +1,13 @@
+// Copyright (C) 2022, Chain4Travel AG. All rights reserved.
```
Comment (c4t-ag) [inline vms/platformvm/blocks/builder/builder.go:1]: I suggest fo make a camino-builder, it is (except tests) only instanciated in vm.go

File: vms/platformvm/blocks/builder/builder.go [hunk 3]
```diff
@@ -392,6 +402,25 @@ func buildBlock(
 		)
 	}
 
+	// TODO@ array of deposits with equal timestamp
+	depositTxID, shouldUnlock, err := builder.getNextDepositToUnlock(timestamp, parentState)
```
Comment (c4t-ag) [inline vms/platformvm/blocks/builder/builder.go:406]: I think we have to compare staker time and deposit time, and use the smaller one.
This impl. priorizes always staker

File: vms/platformvm/blocks/executor/verifier.go [hunk 4]
```diff
@@ -265,15 +265,15 @@ func (v *verifier) banffNonOptionBlock(b blocks.BanffBlock) error {
 		)
 	}
 
-	nextStakerChangeTime, err := executor.GetNextStakerChangeTime(parentState)
```
Comment (c4t-ag) [inline vms/platformvm/blocks/executor/verifier.go:268]: I'm not a friend in renaming, if we overwrite builderf.go, these changes can be omitted

File: vms/platformvm/config/camino.go [hunk 5]
```diff
@@ -3,6 +3,9 @@
 
 package config
 
+import "time"
+
 type CaminoConfig struct {
-	DaoProposalBondAmount uint64
+	DaoProposalBondAmount     uint64
+	GradualUnlockHalfDuration time.Duration
```
Comment (c4t-ag) [inline vms/platformvm/config/camino.go:10]: Same as in 1st comment, better have this in rule

**Review by c4t-ag — CHANGES_REQUESTED** [review]

File: vms/platformvm/blocks/builder/builder.go (hunk 3 above)
Comment (evlekht) [reply vms/platformvm/blocks/builder/builder.go:406]: I intended to do as u said, probably just didn't implement yet

**Pull request closed without merging**